### PR TITLE
feat: HQM-27 sign contarct white list

### PR DESCRIPTION
**Pull request by iGroza**
Fixes #

## Proposed Changes

  - Closed: HQM-27 

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/debug-vars.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ const debug: DebugVars = {
   enableSkipPinOnLogin: false,
   enableWeb3BrowserLogger: true,
   allowAnySourcesForWalletConnectLogin: false,
-  disableWeb3DomainBlocking: false,
+  disableWeb3DomainBlocking: true,
```
Comment (ragozin-nikita) [inline src/debug-vars.ts:35]:
```suggestion
  disableWeb3DomainBlocking: false,
```

File: src/utils.ts [hunk 2]
```diff
@@ -719,3 +723,48 @@ export async function fetchWithTimeout(
 }
 
 export const decimalToHex = (value: string) => new Decimal(value).toHex();
+
+export const getTransactionFromJsonRpcRequest = (
+  request: PartialJsonRpcRequest,
+): Partial<JsonRpcTransactionRequest> | undefined => {
+  if (
+    request.method === EIP155_SIGNING_METHODS.ETH_SIGN_TRANSACTION ||
```
Comment (ragozin-nikita) [inline src/utils.ts:731]:
```suggestion
[EIP155_SIGNING_METHODS.ETH_SIGN_TRANSACTION, EIP155_SIGNING_METHODS.ETH_SEND_TRANSACTION]
.includes(request.method)
```

File: src/utils.ts [hunk 3]
```diff
@@ -719,3 +723,48 @@ export async function fetchWithTimeout(
...
+  } catch (err) {
+    Logger.error('verifyAddress', err);
+    return null;
+  }
+};
+
+export function isContractTransaction(
+  tx: JsonRpcTransactionRequest | undefined,
+): boolean {
+  if (!tx) {
```
Comment (ragozin-nikita) [inline src/utils.ts:758]:
```suggestion
  if (!tx || !tx.to) {
```

File: src/utils.ts [hunk 4]
```diff
@@ -719,3 +723,48 @@ export async function fetchWithTimeout(
...
+    return false;
+  }
+  if (!tx.to) {
```
Comment (ragozin-nikita) [inline src/utils.ts:761]:
```suggestion
```

File: src/screens/json-rpc-sign-screen.tsx [hunk 5]
```diff
@@ -65,11 +77,30 @@ export const JsonRpcSignScreen = () => {
...
     }
   }, [chainId, isAllowed, navigation, onPressReject, request, wallet]);
 
+  const checkContractAddress = useCallback(async () => {
+    if (isTransaction) {
+      const params = getTransactionFromJsonRpcRequest(request);
+
+      if (params?.from) {
+        const info = await verifyAddress(params.from);
+        info && setVerifyAddressResponse(info);
```
Comment (ragozin-nikita) [inline src/screens/json-rpc-sign-screen.tsx:89]: Do not create boolean variable here
```suggestion
if (info) setVerifyAddressResponse(info);
```

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

**Review by vivalaakam — APPROVED** [review]

**Review by ragozin-nikita — APPROVED** [review]

**Pull request merged**